app/src/main/kotlin/com/intive/picover/common/viewmodel/MVIViewModel.kt

```diff
@@ -10,34 +10,39 @@ import com.intive.picover.common.viewmodel.state.MVIState
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.launch
 
-abstract class MVIViewModel<S : MVIState, E : Event>(initialState: S) : ViewModel() {
+abstract class MVIViewModel<S : MVIState, E : Event, SideEffect : Any>(initialState: S) : ViewModel() {
 
 	@Suppress("ktlint:standard:property-naming")
 	protected val _state = mutableStateOf(initialState)
 	val state: State<S> = _state
-	private val events: MutableSharedFlow<E> = MutableSharedFlow()
+	val sideEffects = MutableSharedFlow<SideEffect>()
```

Comment:
as we already have private mutable field for `state` maybe it's good idea to do the same for `sideEffects` as it should only change in vm?

Reply:
Good point, I'll improve that when rest of the feedback will be posted.